File: backend/app/src/models/population/Worker.py
from ..blueprints.Human import Human


class Worker(Human):
    def __init__(self, name: str, employed: bool = False, field_of_work = "unskilled", working: bool = False, production: int = 0):
        super().__init__(name=name, profession="worker")
        self.employed: bool = employed
        self.field_of_work: str = field_of_work
        self.working: bool = working
        self.production: int = production

    def fetch_population_state(self) -> dict:
        population_state: dict = super().fetch_population_state()
        population_state.update({
            "employed": self.employed,
            "field_of_work": self.field_of_work,
            "working": self.working,
            "production": self.production
        })
        
        return population_state





if __name__ == "__main__":
    pass

File: backend_refactoring/CSharpApi/Models/BluePrints/Beings/Worker.cs
namespace CSharpApi.Models.BluePrints.Beings
{
    public class Worker : Human
    {
        public string WorkerName { get; set; } = string.Empty;
        public bool Employed { get; set; } = false;
        public string FieldOfWork { get; set; } = string.Empty;

        // public bool Working = false;

        public bool Alive { get; set; } = true;

        public int Production { get; set; } = 0;

        // public string Profession { get; set; } = "worker";

        public Worker(string workerName,
            bool employed = false,
            string profession = "worker",
            bool alive = true,
            string fieldOfWork = "unskilled",
            bool working = false,
            int production = 0) 
            : base(workerName, profession, alive)
        {
            WorkerName = workerName;
            Profession = profession;
            Employed = employed;
            Alive = alive;
            FieldOfWork = fieldOfWork;
            Working = working;

        }
    }
}
